python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/launcher_tests/test_container_context.py

```diff
@@ -70,3 +79,7 @@ def test_merge(
         empty_container_context.merge(secrets_container_context).secrets_tags
         == secrets_container_context.secrets_tags
     )
+
+    merged = merged.merge(environment_container_context)
```

Comment:
This isn't really testing any merge behaviors. I'd probably modify the existing container context fixtures to include environment. And then assert things like:

- what happens if I merge a container context with environment variable `FOO` into a container context without that environment variable?
- what happens if I merge a container context with environment variable `FOO` into a container context without any environment variables?
- what happens if I merge a container context with environment variable `FOO` into a container context that already has environment variable `FOO`? Which one takes precedence?
- what happens if I merge a container context without environment variable `FOO` into a container context with environment variable `FOO`?

Reply:
Looks like an edge case that wasn't tested for when container context was added for secrets. @gibsondan is this behavior intentional?

I agree that we probably want to change the behavior (and perhaps to choose the "newer" (second) one if there is a collision - whichever direction we go, we should include the test you wrote so we can detect regressions if somebody were to change that behavior in the future.

Reply:
in general when there's a collision the intention is that the other one being passed in wins, yeah. I'd be down for making that improvement here